### ISSUE TITLE
Incorrect crate feature detection

### ISSUE THREAD

**Issue by aditsachde** [rust-analyzer/rust-analyzer#6504]
I'm using `stm32f3xx-hal`, a crate that depends heavily on feature flags to provide board-specific attributes. However, rust-analyzer does not properly pick up on these. I added the most minimal example I could come up with. Compile using `rustup target add thumbv7em-none-eabihf && cargo build --target thumbv7em-none-eabihf`.

All the imports under `gpioa, gpiob, gpioe` are based on the feature flags, and rust-analyzer will show an error, however, it does compile correctly.

`main.rs`
```#![no_std]
#![no_main]

use cortex_m_rt::entry;
use panic_halt as _;
use stm32f3xx_hal::gpio::{
    gpioa::{PA10, PA5, PA6, PA7, PA9},
    gpiob::{PB13, PB14, PB15},
    gpioe::{PE3, PE4},
};

#[entry]
fn main() -> ! {
    loop {}
}
```

`Cargo.toml`
```
[package]
name = "testcrate"
version = "0.1.0"
authors = ["Adit Sachde"]
edition = "2018"

[dependencies]
panic-halt = "0.2.0"
cortex-m-rt = "0.6.10"

[dependencies.stm32f3xx-hal]
features = ["stm32f303xc"]
version = "0.5.0"
```

**Comment by bjorn3** [comment]
Does it error when compiling with `--all-targets --all-features`?

**Comment by aditsachde** [comment]
Yes. The `cortex-m-rt` crate, which is also a dependency of `stm32f3xx-hal` does not compile on x86-64, as it provides runtime functionality for the Cortex M Arm-based series of processors.

**Comment by jonas-schievink** [comment]
This now produces a diagnostic in the stm32f3xx-hal crate:

> Total tokens count exceed limit : count = 355617

**Comment by jonas-schievink** [comment]
Fix in #6804, which just bumps the token limit
